integration/singularity/store.go

```diff
@@ -417,8 +419,52 @@ func (s *SingularityStore) Put(ctx context.Context, reader io.ReadCloser) (*blob
 	return desc, nil
 }
 
+func (s *SingularityStore) PassGet(w http.ResponseWriter, r *http.Request, id blob.ID) {
+	idStream, err := os.Open(filepath.Join(s.local.Dir(), id.String()+".id"))
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			logger.Warnw("Cannot find requested id file", "id", id.String())
+			http.Error(w, "", http.StatusNotFound)
+			return
+		}
+		logger.Errorw("Cannot open id file", "err", err, "id", id.String())
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+	fileIDString, err := io.ReadAll(idStream)
+	if err != nil {
+		logger.Errorw("Cannot read id file", "err", err, "id", id.String())
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+	fileID, err := strconv.ParseUint(string(fileIDString), 10, 64)
+	if err != nil {
+		logger.Errorw("Cannot parse id file", "err", err, "id", id.String())
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+
+	rangeHeader := r.Header.Get("Range")
+
+	_, _, err = s.singularityClient.File.RetrieveFile(&file.RetrieveFileParams{
+		Context: r.Context(),
+		ID:      int64(fileID),
+		Range:   ptr.String(rangeHeader),
+	}, w)
+	if err != nil {
+		logger.Errorw("Failed to retrieve file slice", "err", err, "id", id.String(), "fileID", fileID)
+		if strings.Contains(err.Error(), "404") {
+			http.Error(w, "", http.StatusNotFound)
+			return
+		}
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+	logger.Info("Retrieved file", "id", id.String())
+}
+
 func (s *SingularityStore) Get(ctx context.Context, id blob.ID) (io.ReadSeekCloser, error) {
```

Comment:
Question: do you think it's better to just return 
```
new Error("not implemented, please use PassGet")
```
so we don't end up maintaining code path that never gets hit

Reply:
I left `Get` because I did not know if there would ever be a case where we want to retrieve a blob from singularity, but we are not doing this int response to a HTTP request. For example, suppose we want to retrieve a file from singularity, and then do something to that file (filter it, compress it, etc.), and then return the result. In that case, we will need the original `Get`.

Reply:
We can always remove it later in another PR if we determine there is no use.